src/devicemanager/devicemanager.cpp

```diff
@@ -97,11 +97,10 @@ void DeviceManager::connectLink(LinkConfiguration* linkConf)
 
     // We could use a single Ping instance, but since we are going to support multiple devices
     // this pointer will hold everything for us
-    if(linkConf->deviceType() == PingDeviceType::PING1D) {
-        _primarySensor.reset(new Ping());
-    } else {
-        _primarySensor.reset(new Ping360());
-    }
+     _primarySensor.reset( linkConf->deviceType() == PingDeviceType::PING1D
+        ? static_cast<PingSensor*>(new Ping())
```

Comment:
my preference is the prior if-else for readability.

Reply:
actually both versions should die, PingDeviceType is an enum, and thus we need to use an switch.

Reply:
ok